### ISSUE TITLE
LinkComponent.extend stops prefixing mount point

### ISSUE THREAD

**Issue by dustinfarris** [ember-engines/ember-engines#304]
As noted in #27, LinkComponent prefixes the mount point to the route.

This isn't the case if you extend LinkComponent though.

```js
// components/custom-link-to.js

export default LinkComponent.extend();
```

```hbs
<!-- Works! -->
{{#link-to "coolness"}}Link to /my-engine/coolness{{/link-to}}

<!-- Doesn't work -->
{{#custom-link-to "coolness"}}Throws "there is no route named coolness"{{/custom-link-to}}
```

This might be an "I'm sorry you can't do that, Dave"—in which case we may want an error handler.

Or maybe move some of `-private/link-to-component` to a public mixin.

**Comment by rwjblue** [comment]
Well, the temp solution is to extend from our modified `LinkComponent`.  By using `Ember.LinkComponent.extend` you are extending from the plain vanilla version which is not aware of mount point.

For custom link-to's within an engine, I think you should do:

```js
import LinkComponent from 'ember-engines/components/link-to';

export default LinkComponent.extend({

});
```

**Comment by dustinfarris** [comment]
@rwjblue i just updated the description.  the modified link component is now in `-private/`

**Comment by rwjblue** [comment]
Sorry, I was just referencing #27 when typing the above. I didn't look at where it was on master.  Ya, we should probably export the component publicly under the ember-engines namespace. 

@ember-engines/maintainers - Thoughts?

**Comment by trentmwillis** [comment]
@rwjblue that would be the simplest solution. Should we also export the link-to-external component in the ember-engines namespace? We currently do it in the app namespace, but that seems bad for extensibility use cases like the one here.

**Comment by rwjblue** [comment]
We should move https://github.com/ember-engines/ember-engines/blob/master/addon/-private/link-to-component.js out of -private, so folks can extend from it like I mentioned above. 

There is still some discussion that we need to have with @tomdale about Ember.LinkComponent being public or not, but either way it's at least intimate API and we should probably have a way to extend it for engines users.

**Comment by rwjblue** [comment]
Closed by #304
